### PR TITLE
UCP/WORKER: add ptr_map

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
add ptr_map

## Why ?
the container will be used in wire protocols to exchange keys instead of raw pointers to EPs and requests

### CODE REVIEW

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -1823,6 +1824,10 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
     ucs_conn_match_init(&worker->conn_match_ctx, sizeof(uint64_t),
                         &ucp_ep_match_ops);
     kh_init_inplace(ucp_worker_rkey_config, &worker->rkey_config_hash);
+    status = ucs_ptr_map_init(&worker->ptr_map);
+    if (status != UCS_OK) {
+        goto err_free;
```
Comment (brminich) [inline src/ucp/core/ucp_worker.c:1829]: err_free assumes `ucs_strided_alloc_cleanup(&worker->ep_alloc);` will be called, while `ucs_strided_alloc_init` was not called yet

Comment (evgeny-leksikov) [reply src/ucp/core/ucp_worker.c]: fixed

**Comment by yosefe** [comment]
NFS issue
bot:pipe:retest

**Comment by evgeny-leksikov** [comment]
> NFS issue
> bot:pipe:retest

thanks! ok to merge now?

**Comment by evgeny-leksikov** [comment]
oops "quote replay" retriggered CI, last testing has been passed.

**Pull request merged**